beacon-chain/db/state.go

```diff
@@ -88,6 +88,37 @@ func (db *BeaconDB) InitializeState(ctx context.Context, genesisTime uint64, dep
 	})
 }
 
+// RiskyHeadState accesses the underlying cached state without cloning it. It is dangerous to call this
+// method if you are doing any write operations to the beacon state.
+func (db *BeaconDB) RiskyHeadState(ctx context.Context) (*pb.BeaconState, error) {
```

Comment:
Not sure if we should even be using this, this is unsafe by default. We would have to mentally make a note whenever we use this method that the cached state isnt being mutated by what we are doing with it